results/pkg/grpc/creds.go

```diff
@@ -0,0 +1,53 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/idtoken"
+	"google.golang.org/grpc/credentials"
+)
+
+// Google provides an authenticated transport for use against Google APIs
```

Comment:
I think we need to do this in a provider-agnostic way from the beginning, or we risk having to refactor all of this before we can promote it to a real top-level project. I'd rather do this without creds, insecurely, than bake in GCP support like this.

Let's keep this code stashed somewhere for later, though, since I think we can use it when the general solution exists.